src/Common/AllocatorWithMemoryTracking.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <stdexcept>
+#include <cstddef>
+#include <cstdlib>
+
+#include <Common/CurrentMemoryTracker.h>
+
+
+/// Implementation of std::allocator interface that tracks memory with MemoryTracker.
+/// NOTE We already plug MemoryTracker into new/delete operators. So, everything works even with default allocator.
+/// But it is enabled only if jemalloc is used (to obtain the size of the allocation on call to delete).
+/// And jemalloc is disabled for builds with sanitizers. In these cases memory was not always tracked.
```

Comment:
Worth enabling only if jemalloc had been disabled? (`USE_JEMALLOC`)

Reply:
I think it's always better - it moves the place where memory is tracked closer to our code. (Tracking memory inside global operator new is more troublesome).